MarkBind/markbind
# Refactor math formulae support

**What is the purpose of this pull request?**




**Overview of changes:**

Related to #1754
With the latest update in [`markdown-it-texmath`](https://github.com/goessner/markdown-it-texmath/issues/40#issuecomment-1140189629), it is now possible to specify a list of delimiters. Hence, this PR aims to refactor the code to take advantage of the new feature.
- Support `dollars` and `brackets` out of the box for math formulae display
- Repurpose the previous `texWithDollars` plugin into `mathDelimiters`, for additional support of other delimiters (optional and only for people who actually need it)
- Update documentation
- Update functional tests
- One slightly unrelated change to print out the error stack trace when a plugin fails to load properly, for a better debugging experience

**Anything you'd like to highlight / discuss:**
nil

**Testing instructions:**
See documentation and functional tests
- https://deploy-preview-1940--markbind-master.netlify.app/userguide/formattingcontents#math-formulae
- https://deploy-preview-1940--markbind-master.netlify.app/userguide/usingplugins#plugin-mathdelimiters

**Proposed commit message: (wrap lines at 72 characters)**
Refactor math formulae support

The `dollars` delimiter for math formulae is currently supported via
a plugin because the markdown-it-texmath package does not
allow for more than one delimiter.

With the latest version update, the markdown-it-texmath package
can now support multiple delimiters.

Let's update the code to specify `dollars` as a supported delimiter
out of the box for math formulae. Let's also repurpose the
`texWithDollars` plugin to `mathDelimiters` for any additional
delimiter addition if any user is interested.

Users can now use `dollars` and `brackets` (two popular) delimiters
for math formulae without any configurations.

---

**Checklist:** :ballot_box_with_check:
